src/core/filesystem.cc

```diff
@@ -916,7 +916,7 @@ ASFileSystem::FileModificationTime(const std::string& path, int64_t* mtime_ns)
   RETURN_IF_ERROR(ParsePath(path, &container, &object_path));
 
   auto blobProperty = bc.get_blob_property(container, object_path);
-  if (!blobProperty.valid()) {
+  if (errno != 0) {
```

Comment:
Very strange for a C++ API to use errno...

Reply:
That is what I thought as well. Additionally there are some cases where the valid is set to false even when errno is set to 0. While I was not able to reproduce such a case this is what the caused the error that the user was seeing in issue 2966